### DISCUSSION TITLE
Trouble with constructing BooleanQuery

### DISCUSSION THREAD

**Discussion by joshuaMarkMuller** [coady/lupyne#19] [General]
I've been tasked to replicate a specific preexisting Lucene search. Currently I can create most of the original Lucene query however I'm having issues constructing the BooleanQuery object.

Currently I'm trying to construct the query this way:
`q1 = Boost(Q.term('text1', searchTerm), value = 2)| Boost(Q.phrase('text1', searchTerm.split(' ')), value = 1.5)| Fuzzy('text1', searchTerm, 1)`
`q2 = Boost(Q.term('text2', searchTerm), value = 2)| Boost(Q.phrase('text2', searchTerm.split(' ')), value = 1.5)| Fuzzy('text2', searchTerm, 1)`
`query = q1 + q2`

Is this formatting possible with Lupyne?

If not, is it possible to create a query from a raw Lucene query syntax as a string?

**Comment by coady** [comment]
I'm not sure what `q1 + q2` is supposed to represent, but here's how to construct the query in lupyne:
```python
In []: from lupyne import engine
In []: Q = engine.Query
In []: searchTerm = 'search term'

In []: Q.boost(Q.term('text1', searchTerm), 2.0) | Q.boost(Q.phrase('text1', searchTerm.split(' ')), 1.5) | Q.fuzzy('text1', searchTerm, 1)
Out[]: <BooleanQuery: ((text1:search term)^2.0 (text1:"(search term)")^1.5) text1:search term~1>
```

**Reply by joshuaMarkMuller** [reply]
Thanks for the response, do you know how to construct a proximity query? For example in parser syntax `text:"search term"~10`

**Reply by coady** [reply]
```python
>>> Q.phrase('text', 'search', 'term', slop=10)
<MultiPhraseQuery: text:"search term"~10>
```
